nixos/modules/config/dotd.nix

```
@@ -0,0 +1,59 @@
+{ config, pkgs, lib, ... }:
+let
+  inherit (lib) mdDoc mkOption mkMerge mkIf mkEnableOption types;
+  cfg = config.environment.dotd;
+in {
+  options = {
+    environment.dotd = mkOption {
+      description = mdDoc "Module to create .d like directories and special files that when read get the contents of the files in the .d folder in sort order";
+      default = {};
+      type = types.attrsOf (types.submodule {
+        options = {
+          enable = mkEnableOption (mdDoc "Create this file, create it's .d folder and the required stuff to make the file get the contents from the files inside the .d folder");
+        };
+      });
+    };
+  };
+  config.systemd = mkMerge (let
+        keys = builtins.attrNames cfg;
+      in builtins.map (key: let
+        item = cfg.${key};
+        normalizedKey = builtins.replaceStrings ["/"] ["_"] key;
+      in {
+        paths."dotd-${normalizedKey}-watcher" = {
+          inherit (item) enable;
+          wantedBy = [ "default.target" ];
+          pathConfig = {
+            PathChanged = "/etc/${key}.d";
+          };
+        };
+        services."dotd-${normalizedKey}-watcher" = {
+          inherit (item) enable;
+          script = ''
+            echo "Reacting to file change..."
+            systemctl restart "dotd-${normalizedKey}" || true
```

```suggestion
            systemctl restart "dotd-${normalizedKey}"
```
Shouldn't this fail here?